syscall_darwin.go

```diff
@@ -7,32 +7,82 @@ import (
 	"unsafe"
 )
 
-func getxattr(path string, name string, value *byte, size int, pos int, options int) (int, error) {
-	r0, _, e1 := syscall.Syscall6(syscall.SYS_GETXATTR, uintptr(unsafe.Pointer(syscall.StringBytePtr(path))), uintptr(unsafe.Pointer(syscall.StringBytePtr(name))), uintptr(unsafe.Pointer(value)), uintptr(size), uintptr(pos), uintptr(options))
-	if e1 != syscall.Errno(0) {
-		return int(r0), e1
+func getxattr(path string, attr string, dest []byte) (sz int, err error) {
+	size := len(dest)
+	var value *byte
+	if size > 0 {
+		value = &dest[0]
 	}
-	return int(r0), nil
+	return darwinGetxattr(path, attr, value, size, 0, 0)
 }
 
-func listxattr(path string, namebuf *byte, size int, options int) (int, error) {
-	r0, _, e1 := syscall.Syscall6(syscall.SYS_LISTXATTR, uintptr(unsafe.Pointer(syscall.StringBytePtr(path))), uintptr(unsafe.Pointer(namebuf)), uintptr(size), uintptr(options), 0, 0)
+func setxattr(path string, attr string, data []byte, flags int) (err error) {
+	size := len(data)
+	var value *byte
+	if size > 0 {
+		value = &data[0]
+	}
+	return darwinSetxattr(path, attr, value, size, 0, flags)
+}
+
+func removexattr(path string, attr string) (err error) {
+	return darwinRemovexattr(path, attr, 0)
+}
+
+func listxattr(path string, dest []byte) (sz int, err error) {
+	size := len(dest)
+	var namebuf *byte
+	if size > 0 {
+		namebuf = &dest[0]
+	}
+	return darwinListxattr(path, namebuf, size, 0)
+}
+
+func parseXattrList(buf []byte) []string {
```

Comment:
?

Reply:
Parsing the data returned from listxattr is os-specific (FreeBSD uses a different format)

Reply:
Yep, I'm just wondering.... Either we just rename it as `toStrings` and move keep both `nullTermToStrings` and `attrListToStrings` in `xattr.go` (just in case) or we just copy-paste  `nullTermToStrings` implementation to _linux_ and _darwin_  and rename freebsd's `attrListToStrings` to `nullTermToStrings`